packages/cln_plugin/lib/src/rpcmethods/manifest/getmanifest.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:cln_plugin/src/rpcmethods/abstractrpcmethod.dart';
+import 'package:cln_plugin/src/rpcmethods/manifest/types/option.dart';
+import 'package:cln_plugin/src/rpcmethods/manifest/types/feature.dart';
+import 'package:cln_plugin/src/rpcmethods/manifest/types/notification.dart';
+
+class ManifestMethod extends AbstractRpcMethod {
+  late List<AbstractRpcMethod> _rpcMethods;
+  late List<String> _subscriptions;
+  late List<Option> _options;
+  late List<String> _hooks;
+  late Feature _features;
+  late bool _dynamic = false;
+  late Set<Notification> _notifications;
+
+  ManifestMethod(String name, String usage, String description)
+      : super("getmanifest", "", "");
+
+  void addFeature(String node, String channel, String init, String invoice) {
+    _features = Feature(node, channel, init, invoice);
+  }
+
+  void addMethods(List<AbstractRpcMethod> methods) {
+    _rpcMethods.addAll(methods);
+  }
+
+  void addMethod(AbstractRpcMethod method) {
+    _rpcMethods.add(method);
+  }
```

Comment:
same here, you hide the "list" implementation with add here.

In addition, are you sure that the rpc method is a List/Set? do you think that it can be a map? `Map<String, AbstractRPCMethod>` where the key is the `method.name`?

Reply:
Map should also do, I thought we would be just storing the RPCMethod, hence the List, also there is no callback implemented yet and it was just metadata so maybe I must have missed something.

Reply:
the map will help to get the call back by name, or in your case the RPCMethod. Core lightning will send you the name of the method that you need to call